### PR TITLE
OGM-1202 Relocate hibernate-ogm-infinispan to hibernate-ogm-infinispan-embedded

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1202

Built on top of https://github.com/hibernate/hibernate-ogm/pull/922

### CODE REVIEW

**Comment by gsmet** [comment]
Pushed a rebase. Reviewing now.

File: infinispan-embedded/pom.xml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        <groupId>org.hibernate.ogm</groupId>
+        <artifactId>hibernate-ogm-parent</artifactId>
+        <version>5.2.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    
+    <artifactId>hibernate-ogm-infinispan-embedded</artifactId>
+    <packaging>jar</packaging>
+
+    <name>Hibernate OGM for Infinispan</name>
```
Comment (gsmet) [inline infinispan-embedded/pom.xml]: "embedded" is missing here?

File: infinispan/pom.xml [hunk 2]
```diff
@@ -22,167 +22,26 @@
...
-                    <skipExec>${skipUnitTests}</skipExec>
-                    <!-- Apache Lucene uses assertions which currently fail on JDK9: -->
-                    <!-- not sure yet how that is going to be resolved, but it's not an OGM problem. -->
-                    <enableAssertions>false</enableAssertions>
-                    <argLine>${additionalRuntimeArgLine}</argLine>
-                    <dependenciesToScan>
-                        <dependency>org.hibernate.ogm:hibernate-ogm-core</dependency>
-                    </dependenciesToScan>
+                    <!-- We need an empty JAR, otherwise the build will fail -->
+                    <skipIfEmpty>false</skipIfEmpty>
```
Comment (gsmet) [inline infinispan/pom.xml]: For Validator, I use the pom packaging for relocation:
https://github.com/hibernate/hibernate-validator/blob/master/relocation/engine/pom.xml
so I don't need to produce a jar.

File: documentation/javadoc/pom.xml [hunk 3]
```diff
@@ -41,7 +41,7 @@
             we import all types from dependant modules -->
         <dependency>
             <groupId>org.hibernate.ogm</groupId>
-            <artifactId>hibernate-ogm-infinispan</artifactId>
+            <artifactId>hibernate-ogm-infinispan-embedded</artifactId>
```
Comment (gsmet) [inline documentation/javadoc/pom.xml]: I think you also need to change the `sourcepath` attribute a little further. And probably add the infinispan-remote module too.

Comment (DavideD) [reply documentation/javadoc/pom.xml]: Good catch!

**Comment by DavideD** [comment]
I've applied the remarks and rebased

**Comment by gsmet** [comment]
Rebased and merged, thanks!

**Pull request closed without merging**